src/widget/wfindonmenu.cpp

```diff
@@ -0,0 +1,218 @@
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(albumName, prefixActionAlbum),
+                this,
+                [this, soundCloudUrl, albumName]() {
+                    emit triggerBrowser(soundCloudUrl, albumName);
+                });
+        break;
+    }
+
+    case Services::LastFm: {
+        const auto lastFmUrl = searchUrlLastFmAlbum();
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(albumName, prefixActionAlbum),
+                this,
+                [this, lastFmUrl, albumName]() {
+                    emit triggerBrowser(lastFmUrl, albumName);
+                });
+        break;
+    }
+
+    case Services::Discogs: {
+        const auto discogsUrl = searchUrlDiscogsGen();
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(albumName, prefixActionAlbum),
+                this,
+                [this, discogsUrl, albumName]() {
+                    emit triggerBrowser(discogsUrl, albumName);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::addActionsTrackTitle(
+        Services serviceTitle, QString trackTitle, QMenu* m_pService) {
+    const QString prefixActionTrackTitle = tr("Title");
+    switch (serviceTitle) {
+    case Services::SoundCloud: {
+        const auto soundCloudUrl = searchUrlSoundCloudTitle();
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(trackTitle, prefixActionTrackTitle),
+                this,
+                [this, soundCloudUrl, trackTitle]() {
+                    emit triggerBrowser(soundCloudUrl, trackTitle);
+                });
+        break;
+    }
+
+    case Services::LastFm: {
+        const auto lastFmUrl = searchUrlLastFmTitle();
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(trackTitle, prefixActionTrackTitle),
+                this,
+                [this, lastFmUrl, trackTitle]() {
+                    emit triggerBrowser(lastFmUrl, trackTitle);
+                });
+        break;
+    }
+
+    case Services::Discogs: {
+        const auto discogsUrl = searchUrlDiscogsGen();
+        m_pService->addAction(
+                actionPrefixSuffixSeparator(trackTitle, prefixActionTrackTitle),
+                this,
+                [this, discogsUrl, trackTitle]() {
+                    emit triggerBrowser(discogsUrl, trackTitle);
+                });
+        break;
+    }
+    default:
+        break;
+    }
+}
+
+void WFindOnMenu::openTheBrowser(const QString& serviceUrl,
+        const QString& query) {
+    QUrlQuery qurlquery;
+    qurlquery.addQueryItem("q", query);
+    QUrl url(serviceUrl);
+    url.setQuery(qurlquery);
+    QDesktopServices::openUrl(url);
+}
+
+void WFindOnMenu::createService(QMenu* serviceMenu,
+        const Track& track,
+        QString serviceTitleDisplay,
+        Services serviceTitle) {
+    serviceMenu = new QMenu(this);
```

Comment:
Since WFindOnMenu is owned by the Q_OBJECT? So something like 
`pserviceMenu = new WFindOnMenu(this);`
would solve this issue?

Reply:
auto pServiceMenu = make_parented<QMenu>(this);

Since WFindOnMenu is inherited from QObject, it will delete all child QObjects once it is deleted, during the inherited QObject destructor.

parented_ptr makes sure that this is the case and will be still the case after refactoring.